fitbenchmarking/controllers/scipy_go_controller.py

```diff
@@ -100,4 +99,6 @@ def cleanup(self):
         else:
             self.flag = 2
 
-        self.final_params = self._popt
+        self.final_params = self._result.x
+        self.iteration_count = self._result.nit
+        self.func_evals = self._result.nfev
```

Comment:
I noticed `self.iteration_count` and `self.func_evals` are sometimes set in the **cleanup** method and sometimes in the **fit** method. I personally like it when it is done in the **cleanup** method. But I know for the Matlab stuff it is easier to do it in the **fit** method because the results are not saved in a class attribute. Do you think it would be better doing it consistently so it is much clearer when a new controller is added ?

Reply:
I did think this as I was working through the different controllers - where possible I have set `self.iteration_count`/`self.func_evals` in cleanup, but as you say for some controllers like the Matlab ones and RALFit this information is not saved in a class attribute so I thought it would be neater to set these within fit rather than create additional class attributes. Happy to change this though as I agree it would make it clearer for writing new controllers if these are always set within cleanup

Reply:
I will leave it up to you to decide how to do the refactoring part. Maybe there are more things that can be optimized and another issue can be created to look at the controllers. But it would be nice to define a standard whatever you feel is best. Because I just noticed it while reviewing but I am not sure how many things you might have to change.

I also noticed that a link was broken in the docs for minuit while reviewing the PR. Maybe it can be fixed here but it is not urgent. It can also be done in another PR.

Reply:
Sounds good - regarding the broken minuit link, I think I opened an issue for this last year so happy to fix it in this PR so that it is actually fixed and not forgotten about again!